### ISSUE TITLE
Pipelines: two builds are triggered instead of one

### ISSUE THREAD

**Issue by ljelinkova** [openshiftio/openshift.io#3742]
I've ran the E2E tests and encountered strange pipelines behavior. 

I created quickstart and ran the pipeline, but two builds were triggered instead of one. The builds are inconsistent between OSIO and Jenkins. 

OSIO: Build 1 has not started at all, build 2 has failed
Jenkins: Build 1 failed, build 2 successful

![pipeline-2](https://user-images.githubusercontent.com/2078072/40973885-1d40e5bc-68c6-11e8-8a25-ca366b6a9b13.png)

And at Jenkins
![jenkins](https://user-images.githubusercontent.com/2078072/40973887-1f20bce0-68c6-11e8-97de-b627253a8645.png)

As you can see the first build in Jenkins failed with the following error. 
```
Failed to execute goal io.fabric8:fabric8-maven-plugin:3.5.38:build (fmp) on project e2e-0605-111854: Failed to execute the build: Unable to build the image using the OpenShift build service: An error has occurred. timeout: Socket closed -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.fabric8:fabric8-maven-plugin:3.5.38:build (fmp) on project e2e-0605-111854: Failed to execute the build
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to execute the build
	at io.fabric8.maven.plugin.mojo.build.BuildMojo.buildAndTag(BuildMojo.java:270)
	at io.fabric8.maven.docker.BuildMojo.executeInternal(BuildMojo.java:44)
	at io.fabric8.maven.plugin.mojo.build.BuildMojo.executeInternal(BuildMojo.java:228)
	at io.fabric8.maven.docker.AbstractDockerMojo.execute(AbstractDockerMojo.java:223)
	at io.fabric8.maven.plugin.mojo.build.BuildMojo.execute(BuildMojo.java:199)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 20 more
Caused by: io.fabric8.maven.core.service.Fabric8ServiceException: Unable to build the image using the OpenShift build service
	at io.fabric8.maven.core.service.openshift.OpenshiftBuildService.build(OpenshiftBuildService.java:121)
	at io.fabric8.maven.plugin.mojo.build.BuildMojo.buildAndTag(BuildMojo.java:267)
	... 26 more
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:62)
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:53)
	at io.fabric8.openshift.client.dsl.internal.BuildConfigOperationsImpl.fromInputStream(BuildConfigOperationsImpl.java:276)
	at io.fabric8.openshift.client.dsl.internal.BuildConfigOperationsImpl.fromFile(BuildConfigOperationsImpl.java:231)
	at io.fabric8.openshift.client.dsl.internal.BuildConfigOperationsImpl.fromFile(BuildConfigOperationsImpl.java:68)
	at io.fabric8.maven.core.service.openshift.OpenshiftBuildService.startBuild(OpenshiftBuildService.java:361)
	at io.fabric8.maven.core.service.openshift.OpenshiftBuildService.build(OpenshiftBuildService.java:111)
	... 27 more
Caused by: java.net.SocketTimeoutException: timeout
	at okio.Okio$4.newTimeoutException(Okio.java:230)
	at okio.AsyncTimeout.exit(AsyncTimeout.java:285)
	at okio.AsyncTimeout$2.read(AsyncTimeout.java:241)
	at okio.RealBufferedSource.indexOf(RealBufferedSource.java:345)
	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:217)
	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:211)
	at okhttp3.internal.http1.Http1Codec.readResponseHeaders(Http1Codec.java:187)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:61)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:125)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.openshift.client.internal.OpenShiftOAuthInterceptor.intercept(OpenShiftOAuthInterceptor.java:66)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:200)
	at okhttp3.RealCall.execute(RealCall.java:77)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:377)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:359)
	at io.fabric8.openshift.client.dsl.internal.BuildConfigOperationsImpl.fromInputStream(BuildConfigOperationsImpl.java:274)
	... 31 more
Caused by: java.net.SocketException: Socket closed
	at java.net.SocketInputStream.read(SocketInputStream.java:204)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at sun.security.ssl.InputRecord.readFully(InputRecord.java:465)
	at sun.security.ssl.InputRecord.read(InputRecord.java:503)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:983)
	at sun.security.ssl.SSLSocketImpl.readDataRecord(SSLSocketImpl.java:940)
	at sun.security.ssl.AppInputStream.read(AppInputStream.java:105)
	at okio.Okio$2.read(Okio.java:139)
	at okio.AsyncTimeout$2.read(AsyncTimeout.java:237)
	... 56 more
```

**Comment by ldimaggi** [comment]
Are you able to recreate this bug manually? It looks like the 2nd build actually ran before the first?

**Comment by piyush1594** [comment]
Generating two pipelines is duplicate of https://github.com/openshiftio/openshift.io/issues/3455 and build failing is duplicate of #3058

**Comment by joshuawilson** [comment]
If this is really a duplicate, we should close it.

**Comment by ljelinkova** [comment]
The #3455  says that the second job is created after clicking the Promote button. I my case I didn't get so far - the build failed and the second one started.

**Comment by jiekang** [comment]
Maybe the 'clicking Promote' is a red herring then. Sounds like both builds are triggered and it will only run one at a time so for a successful build, it will not finish till Promote is clicked. Then the second build starts. 

In your case for a failed build, the second will start then.

**Comment by ldimaggi** [comment]
In Lucia's case - I think that the 2nd build ran before the first.   ;-)

**Comment by jiekang** [comment]
Probably best to keep one issue for tracking this two builds queued problem though. I've closed #3455 in favor of this. @ljelinkova can you update this issue a little to make it clear the problem is that two builds are being triggered instead of one.

**Comment by jiekang** [comment]
Yeah, in my issue, build 2 also ran before build 1.

**Comment by jiekang** [comment]
I also saw this when running E2E smoke tests locally, causing them to fail.

**Comment by piyush1594** [comment]
This is an intermittent issue and happens sometimes. cc @sthaha

**Comment by sthaha** [comment]
What I observed was 

1. Pipeline starts with `Build - 2` instead of `Build - 1`
1. The `Build - 2` fails and then `Build - 1` starts 

OSO pipeline view looks a below: 
![2-pipelines](https://user-images.githubusercontent.com/3005132/41886566-3d89a1b2-7940-11e8-9ec7-2a8ed1e57ff3.png)


Build - 2 fails with the following error
```
[INFO] Current reconnect backoff is 32000 milliseconds (T5)
[WARNING] Executor didn't terminate in time after shutdown in close(), killing it in: io.fabric8.kubernetes.client.dsl.internal.WatchHTTPManager@1c4dcd0
[ERROR] F8: Failed to execute the build [Unable to build the image using the OpenShift build service]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 03:47 min
[INFO] Finished at: 2018-06-26T02:54:38+00:00
[INFO] Final Memory: 42M/78M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.fabric8:fabric8-maven-plugin:3.5.38:build (fmp) on project st-vertx-booster-j26: Failed to execute the build: Unable to build the image using the OpenShift build service: OpenShift Build st-vertx-booster-j26-s2i-1: Failed: PullBuilderImageFailed -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.fabric8:fabric8-maven-plugin:3.5.38:build (fmp) on project st-vertx-booster-j26: Failed to execute the build
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
```


It seems to me that launcher is starting two pipelines instead of one which can be verified by going to openshift console and starting two pipelines.

**Comment by piyush1594** [comment]
@gastaldi I am trying to reproduce this issue and not able to do so. Can you please help me regarding this. I checked the repo https://github.com/sthaha/st-vertx-booster-j26 there is only single commit by launchpad. Can you please guide so that i can reproduce this cc @ravsa @animuk

**Comment by piyush1594** [comment]
I checked https://github.com/ljelinkova-testing/e2e-0605-111854 also, there is a single commit too.

**Comment by gastaldi** [comment]
@piyush1594 @ravsa can you try running the launcher backend locally ? I wonder if the order of the steps declared in https://github.com/fabric8-launcher/launcher-backend/blob/master/addons/osio-addon/src/main/java/io/fabric8/launcher/osio/OsioLaunchMissionControl.java needs to be changed

**Comment by piyush1594** [comment]
@gastaldi @ravsa What I have observed so far is that the BuildConfig Created has property lastVersion in status equal to 2 in case of two builds of a pipeline. Have you any idea where that is getting changed. If that property is 1 it shows one build.

**Comment by gastaldi** [comment]
I remember @aslakknutsen saying something that the order in which the steps are executed needed to be changed, but I don't remember what was the issue back then, maybe this problem may be related

**Comment by piyush1594** [comment]
@gastaldi last version in build config yaml changes because of webhook event, I just checked by pushing a commit, it increments the last version in yaml and starts new build there in the pipeline. So seems like something is happening in between the process because I checked that in case of the 2-3 pipeline also there is only a single commit in the repo.

**Comment by piyush1594** [comment]
@gastaldi Are you talking about this https://github.com/openshiftio/openshift.io/issues/3252#issuecomment-384874867

Is this affecting the flow? cc @aslakknutsen @ravsa

**Comment by piyush1594** [comment]
This issue has been fixed in launcher backend using https://github.com/fabric8-launcher/launcher-backend/pull/478 and is in prod by https://github.com/openshiftio/saas-launchpad/pull/178

Feel free to reopen it if you face it next time. Thanks

**Comment by ldimaggi** [comment]
* I am able to reproduce this issue ~100% of the time when running a test directly through the API with a user account provisioned on the starter-us-east-2 cluster.
* I am also able to reproduce this issue ~10-15% of the time when running a test through the UI with a user account provisioned on the starter-us-east-2 cluster.

**Comment by gastaldi** [comment]
@piyush1594 sure, by all means

**Comment by ldimaggi** [comment]
I am also seeing this on cluster starter-us-east-2a - when a test accesses the API directly:

![screenshot from 2018-07-16 14-25-34](https://user-images.githubusercontent.com/642621/42776181-44635f7c-8904-11e8-9044-594473bb9d8b.png)

**Comment by piyush1594** [comment]
Hey, @ldimaggi Can you please check the number of webhook events on the repo.

**Comment by ldimaggi** [comment]
Yes - there is 1 event:

![screenshot from 2018-07-19 14-58-45](https://user-images.githubusercontent.com/642621/42964392-e1afde56-8b64-11e8-8a8d-18a68d229b3a.png)

**Comment by ldimaggi** [comment]
Seeing this problem again today - but - this time when a new app is created in the launcher - not during an import operation - steps to recreate:

* Reset user env
* After confirming that Jenkins and Content Repo are up and running
* Create a new space
* In that space, create a new app (I used REST/Vert.X) in the launcher
* Observe (2) pipelines started:
![screenshot from 2018-07-24 08-13-55](https://user-images.githubusercontent.com/642621/43138171-0792297a-8f1c-11e8-856a-d8ca30199e1f.png)

* Observe (2) events for the corresponding github repo:
![screenshot from 2018-07-24 08-14-55](https://user-images.githubusercontent.com/642621/43138211-1f124de6-8f1c-11e8-9188-4de92760ff2f.png)

**Comment by piyush1594** [comment]
@bartoszmajsak This is still not fixed, reopening it. That PR is still reproducing it if call to push code takes time and create webhook happens before that.

**Comment by bartoszmajsak** [comment]
@piyush1594 Interesting that it has been closed automatically through your commit on my fork :D The only thing I can think of is syncing my forked `master` with upstream. And it seems back then it was also closed. 

Interesting to see that `Fixes #...` on the fork causes the same effect.

**Comment by bartoszmajsak** [comment]
> This is still not fixed, reopening it. That PR is still reproducing it if call to push code takes time and create webhook happens before that.

@piyush1594 As those two are dependent - can this be executed in a sequence? What are the latest findings from your work with @tinakurian?

**Comment by ljelinkova** [comment]
This issue started to occur on pretty frequent basis approximately in the last 7 days. At first, the build starts in OSIO as number 2 but once it is finished the OSIO UI changes to build number 1. 

Are there any additional information, logs etc. that we could provide?
These are screenshots from e2e tests:

1. Build number 2 is started
![05-01-pipeline-info](https://user-images.githubusercontent.com/2078072/46080090-3e0d0d80-c199-11e8-8229-8fc4cbd23153.png)

2. Build number 2 is finished

![05-05-pipeline-finished](https://user-images.githubusercontent.com/2078072/46080111-4e24ed00-c199-11e8-9777-d4407f64108b.png)

3. Jenkins log
http://artifacts.ci.centos.org/devtools/e2e/devtools-test-e2e-openshift.io-smoketest-us-east-2-released/753/05-06-jenkins-log.html
![05-06-jenkins-log](https://user-images.githubusercontent.com/2078072/46080159-701e6f80-c199-11e8-9164-a35cc2b052da.png)

4. OS console
![05-07-os-pipeline](https://user-images.githubusercontent.com/2078072/46080173-7dd3f500-c199-11e8-97aa-21964ab3d4e7.png)

5. OSIO Pipelines screen
![05-08-aftereach](https://user-images.githubusercontent.com/2078072/46080189-86c4c680-c199-11e8-9a2b-9e779045c328.png)


These are some of oc logs
http://artifacts.ci.centos.org/devtools/e2e/devtools-test-e2e-openshift.io-smoketest-us-east-2-released/753/oc-logs-output.txt

**Comment by ppitonak** [comment]
It's happening quite often, another example https://ci.centos.org/job/devtools-test-e2e-openshift.io-smoketest-us-east-2a-released/781/console

**Comment by ppitonak** [comment]
@joshuawilson can somebody from your team have a look?

**Comment by joshuawilson** [comment]
I've added a story to track the e2e bugs at https://openshift.io/openshiftio/Openshift_io/plan/detail/802

**Comment by rupalibehera** [comment]
@tinakurian , are you guys working on this one, this is a P1 and is important for the e2e tests. and we also suspect that it may fix https://github.com/openshiftio/openshift.io/issues/4172 as a side effect which is P1 too. https://github.com/openshiftio/openshift.io/issues/4172 is blocked until https://github.com/openshiftio/openshift.io/issues/3742 is done.

cc: @pgarg @pradeepto @sthaha

**Comment by ppitonak** [comment]
@tinakurian what is the progress on this issue?

**Comment by tinakurian** [comment]
@ppitonak As mentioned, i was not actively working on this issue. I have started to look into this with @ldimaggi help to setup a more automated way to reproduce this issue to help with implementing and testing a fix.

**Comment by ppitonak** [comment]
Happened on Sunday on prod-preview
https://ci.centos.org/job/devtools-test-e2e-prod-preview.openshift.io-smoketest-pr-us-east-2a-released/3494/

**Comment by ppitonak** [comment]
@tinakurian @stevengutz can you please post update? It's a P1 issue which probably causing also #4172

**Comment by sbose78** [comment]
@ppitonak I'm on this as well. Will let you know if I find something.

**Comment by ppitonak** [comment]
@sbose78 any updates?

**Comment by ljelinkova** [comment]
The issue happeneded again 

https://ci.centos.org/job/devtools-test-e2e-openshift.io-smoketest-us-east-1a-beta/1633/console

**Comment by sbose78** [comment]
> unassigning @piyush-garg , we have communicated that, this issue is related to launcher/app-creation

@sthaha @rupalibehera @tinakurian 
Could you please add the above context to this issue in comments so that someone else could pick this up?

**Comment by piyush-garg** [comment]
@sbose78 

```
my observation is https://github.com/fabric8-launcher/launcher-backend/blob/master/addons/osio-addon/src/main/java/io/fabric8/launcher/osio/OsioLaunchMissionControl.java#L74 step of pushing should be completed successfully before moving forward to the next step of creating webhook. In my case, I see two GitHub webhook events, one of pushing code which should never be there as webhook creation is after pushing code. So my answer is it is happening at the condition when codebase pushing take some time and next step webhook creation happens first.
```

Hope this helps. Thanks

**Comment by ldimaggi** [comment]
That explains the intermittent nature of the issue.

**Comment by ljelinkova** [comment]
I've just seen this issue on prod-preview again

http://artifacts.ci.centos.org/devtools/e2e/devtools-test-e2e-prod-preview.openshift.io-smoketest-pr-us-east-2a-beta/4915/05-01-pipeline-info.png

**Comment by piyush-garg** [comment]
@sbose78 @tinakurian 

I got the rights for the repo created above from @ljelinkova and confirmed the same. There are two webhook events - and code push event happened after 3 mins of other.

Thanks @ljelinkova

**Comment by piyush-garg** [comment]
I think my comments will not make sense anymore after this https://github.com/fabric8-launcher/launcher-backend/pull/612

**Comment by piyush-garg** [comment]
This is happening very frequently after https://github.com/fabric8-launcher/launcher-backend/pull/612 now

Actually, Launcher-backend was like this only before as current after PR https://github.com/fabric8-launcher/launcher-backend/pull/612.

I raised the PR https://github.com/fabric8-launcher/launcher-backend/pull/478 that make it very intermittent but I have already explained the fix we need to add to fix this.

Now after the PR-https://github.com/fabric8-launcher/launcher-backend/pull/612 it is basically reverting my PR https://github.com/fabric8-launcher/launcher-backend/pull/478 and it is reproducible very frequently now.

Please have a look at it on priority. This is pending from last 6-7 months.
cc @sbose78 @tinakurian

**Comment by sbose78** [comment]
@piyush-garg , in that case, I think the first step would be to revert https://github.com/fabric8-launcher/launcher-backend/pull/612 ? Is that what you are implying?

**Comment by ldimaggi** [comment]
I started seeing a variation on this problem yesterday. What I am seeing is that when the first build completes, a second build is immediately started.

However, I am not yet able to reliably recreate this behavior as my user account is provisioned on the starter-us-east-2 cluster - and is blocked by these issues:

https://github.com/openshiftio/openshift.io/issues/4668

and

I also am seeing about a 90% failure in Jenkins starting after an env reset:
https://github.com/openshiftio/openshift.io/issues/4451

In trying to build an existing booster, seeing the idler intermittently stop active build pipelines:
https://github.com/openshiftio/openshift.io/issues/4451

**Comment by sbose78** [comment]
@piyush-garg , When webhooks are setup, Github tests out the URL by sending an event. That would explain why their's always an extra webhook event being triggered. Given that there's nothing that could be done on the launcher side to avoid that, could we find a way to ignore that webhook event on Jenkins's side?

**Comment by piyush-garg** [comment]
@sbose78 According to me, the issue is not the `ping` event but a `push` event which happens sometimes. I have observed that in the case of two builds, `push` event is there which is not in case of one build. The reason is as you see here https://github.com/fabric8-launcher/launcher-backend/blob/master/addons/osio-addon/src/main/java/io/fabric8/launcher/osio/OsioLaunchMissionControl.java#L76, event flow is `creating webhook, pushing code, and triggering build` and if push code triggers an event then trigger build creates the second build.

**Comment by ppitonak** [comment]
@piyush-garg @tinakurian If I understand it correctly, you know the root cause. What is the next step? I would appreciate regular updates.

@stevengutz @kpiwko

**Comment by sbose78** [comment]
@gastaldi any thoughts on Piyush's comment above? Any suggestions on what the possible fix could be? The last one hadn't worked.

**Comment by gastaldi** [comment]
@sbose78 one thing that it could be tried (it requires someone to test it though) is to remove the explicit call to trigger the build in launcher-backend (like in the unmerged PR https://github.com/fabric8-launcher/launcher-backend/pull/637)

**Comment by rupalibehera** [comment]
@gastaldi , it would be nice if you could push this to prod-preview so that we can test it.

**Comment by sbose78** [comment]
I'm good with https://github.com/fabric8-launcher/launcher-backend/pull/637 , let's merge this.

**Comment by jarifibrahim** [comment]
fabric8-launcher/launcher-backend#637 removed the second call to trigger build but now I don't see any builds. This seems to be a bug on jenkins-proxy. @piyush-garg can you confirm this?

**Comment by jarifibrahim** [comment]
> @jarifibrahim Check webhook events, if there is no webhook event, then it will not trigger the build

I do have a webhook event.

**Comment by jarifibrahim** [comment]
This issue was fixed by https://github.com/fabric8-launcher/launcher-backend/pull/637 and should no longer be reproducible. 

@ljelinkova can you confirm if this issue is fixed?

**Comment by ppitonak** [comment]
@jarifibrahim due to the nature of this bug, it's not possible to reliably confirm if this issue is fixed. I'll close now and reopen if it appears again.
